front/src/common/OSRDMenu.tsx

```diff
@@ -0,0 +1,88 @@
+import { useEffect, useRef } from 'react';
+
+import cx from 'classnames';
+
+import useModalFocusTrap from 'utils/hooks/useModalFocusTrap';
+
+type MenuOption = {
+  label: string;
+  disabled?: boolean;
+  icon?: React.ReactNode;
+};
+
+type OSRDMenuProps = {
+  width: string;
+  openerRef: React.RefObject<HTMLButtonElement>;
+  items: MenuOption[];
+  onItemClick: (label: string) => void;
```

Comment:
I believe it would be better for each item to handle its own action rather than using a global handler for all items. Since each option has a distinct action, keeping a global handler forces us to manage different cases in the same function, making the code less flexible and less readable (e.g., using if or switch statements).

Reply:
That's much better, you're right. Done.